apps/ensapi/src/lib/ensanalytics/referrer-leaderboard/get-referrer-leaderboard.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  buildReferralProgramRules,
+  buildReferrerLeaderboard,
+  buildReferrerMetrics,
+  type ReferrerLeaderboard,
+  type USDQuantity,
+} from "@namehash/ens-referrals";
+
+import { type AccountId, deserializeDuration, type UnixTimestamp } from "@ensnode/ensnode-sdk";
+
+import logger from "@/lib/logger";
+
+import { getReferrerLeaderboardRecords } from "./database";
+
+/**
+ * Builds a `ReferralLeaderboard` from all database records in the `registrar_actions` table
+ * matching the provided filters.
```

Comment:
```suggestion
 * Builds a `ReferralLeaderboard` from the provided params.
```

Goal: This function shouldn't concern itself with database internals.